lib/rom/processor/transproc.rb

```diff
@@ -99,6 +99,8 @@ def visit(attribute)
       def visit_attribute(attribute)
         if attribute.typed?
           t(:map_value, attribute.name, t(:"to_#{attribute.type}"))
+        elsif attribute.meta[:coercer]
+          t(:map_value, attribute.name, attribute.meta[:coercer])
         end
```

Comment:
the error should be raised in attribute dsl. ie `raise ArgumentError, "can't specify type and block at the same time" if options[:type] && block` or something like that